test/app-luatest/gh_8613_new_cli_behaviour_test.lua

```diff
@@ -0,0 +1,106 @@
+local t = require('luatest')
+local treegen = require('test.treegen')
+local justrun = require('test.justrun').tarantool
+
+local g = t.group()
+
+local misuse = {
+    output = "Invalid usage:"
+          .. " please either provide a Lua script name"
+          .. " or specify an instance name to be started"
+          .. " or set -i CLI flag to spawn Lua REPL.",
+    pattern = false,
+}
+local greeting = {
+    output = "Tarantool [^\n]*\ntype 'help' for interactive help",
+    pattern = true,
+}
+local prompt = {
+    output = 'tarantool>',
+    pattern = false,
+}
+
+local script = string.dump(function() print(_VERSION) end)
+
+local rundir
+g.before_each(function()
+    treegen.init(g)
+    treegen.add_template(g, '^script%.lua$', script)
+    rundir = treegen.prepare_directory(g, {'script.lua'})
+end)
+
+local function simplerun(args)
+    return justrun(rundir, {}, args, {nojson = true, stderr = true})
+end
+
+g.test_tarantool_no_arg_run = function()
+    local result = simplerun({})
+    t.assert_equals(result.exit_code, 1,
+                    'Running Tarantool w/o CLI arguments fails')
+    t.assert_str_contains(result.stderr, misuse.output, misuse.pattern,
+                          'Running Tarantool w/o CLI arguments yields usage')
+end
+
+g.test_tarantool_stdin_invalid_redirect = function()
+    local result = simplerun({'</dev/null'})
+    t.assert_equals(result.exit_code, 1,
+                    'Redirect to /dev/null w/o dash in CLI args')
```

Comment:
to -> from?

Reply:
OK, if it makes the message clearer. Fixed.